libs/blocks/merch/merch.js

```diff
@@ -1,57 +1,152 @@
-import { loadScript, getConfig, createTag } from '../../utils/utils.js';
+import {
+  loadScript,
+  getConfig,
+  createTag,
+} from '../../utils/utils.js';
 
-const { miloLibs, codeRoot, env } = getConfig();
-const base = miloLibs || codeRoot;
+const tacocatVersion = '1.12.0';
+const wcs = { apiKey: 'wcms-commerce-ims-ro-user-milo' };
+const envProd = 'prod';
+const ctaPrefix = /^CTA +/;
 
-function buildButton(a, osi) {
-  if (!a) return null;
+let initialized = false;
+
+// supported languages for the price literals
+const supportedLanguages = [
+  'ar', 'bg', 'cs', 'da', 'de', 'en', 'es', 'et', 'fi', 'fr', 'he', 'hu', 'it', 'ja', 'ko',
+  'lt', 'lv', 'nb', 'nl', 'pl', 'pt', 'ro', 'ru', 'sk', 'sl', 'sv', 'tr', 'uk', 'zh_CN', 'zh_TW',
+];
+
+function omitUndefined(target) {
+  if (target != null) {
+    Object.entries(target).forEach(([key, value]) => {
+      if (value == null) delete target[key];
+    });
+  }
+  return target;
+}
+
+const getTacocatEnv = (envName, locale) => {
+  const scriptUrl = envName === envProd
+    ? `https://www.adobe.com/special/tacocat/lib/${tacocatVersion}/tacocat.js`
+    : `https://www.stage.adobe.com/special/tacocat/lib/${tacocatVersion}/tacocat.js`;
+  // eslint-disable-next-line prefer-const
+  let [language, country = 'us'] = locale.split('-', 2);
```

Comment:
I'm not sure if the country should be 'us' for mena countries. I took a quick look on adobe.com and they also have different prices served to them.

Reply:
good catch, addressed.